plugins/googleanalytics/src/org/pentaho/di/trans/steps/googleanalytics/GaInputStep.java

```diff
@@ -0,0 +1,266 @@
+          if ( GaInputStepMeta.PROPERTY_DATA_SOURCE_ACCOUNT_NAME.equals( fieldName ) ) {
+            // We expect a single account name, and already fetched it during init
+            dataObject = accountName;
+          } else {
+            dataObject = data.feed.getProfileInfo().get( removeClassifier( fieldName ) );
+          }
+        } else if ( GaInputStepMeta.FIELD_TYPE_DATA_SOURCE_FIELD.equals( type ) ) {
+          // Get tableId or tableName
+          if ( GaInputStepMeta.FIELD_DATA_SOURCE_TABLE_ID.equals( fieldName ) ) {
+            dataObject = data.feed.getProfileInfo().get( removeClassifier( fieldName ) );
+          } else {
+            // We only have two Data Source Fields and they're hard-coded, so we handle tableName in this else-clause
+            // since tableId was done in the if-clause. We have to handle the two differently because tableName is
+            // actually the profile name in this version (v3) of the Google Analytics API.
+            dataObject = data.feed.getProfileInfo().getProfileName();
+          }
+        } else if ( GaInputStepMeta.DEPRECATED_FIELD_TYPE_CONFIDENCE_INTERVAL.equals( type ) ) {
+          dataObject = null;
+          if ( log.isRowLevel() ) {
+            logRowlevel( BaseMessages.getString( PKG, "GoogleAnalytics.Warn.FieldTypeNotSupported",
+              GaInputStepMeta.DEPRECATED_FIELD_TYPE_CONFIDENCE_INTERVAL ) );
+          }
+        } else {
+          // Assume it's a Dimension or Metric, we've covered the rest of the cases above.
+          dataObject = entry.get( j++ );
+        }
+        outputRow[i] = data.outputRowMeta.getValueMeta( i ).convertData( data.conversionMeta[i], dataObject );
+      }
+
+      // copy row to possible alternate rowset(s)
+      putRow( data.outputRowMeta, outputRow );
+
+      // Some basic logging
+      if ( checkFeedback( getLinesWritten() ) ) {
+        if ( log.isBasic() ) {
+          logBasic( "Linenr " + getLinesWritten() );
+        }
+      }
+      return true;
+
+    } else {
+      setOutputDone();
+      return false;
+    }
+  }
+
+  protected Analytics.Data.Ga.Get getQuery( Analytics analytics ) {
+
+    Analytics.Data dataApi = analytics.data();
+    Analytics.Data.Ga.Get query;
+
+    try {
+      query = dataApi.ga().get(
+        meta.isUseCustomTableId() ? environmentSubstitute( meta.getGaCustomTableId() ) : meta.getGaProfileTableId(), //ids
+        environmentSubstitute( meta.getStartDate() ), // start date
+        environmentSubstitute( meta.getEndDate() ), // end date
+        environmentSubstitute( meta.getMetrics() ) // metrics
+      );
+      query.setDimensions( environmentSubstitute( meta.getDimensions() ) );
+
+
+      if ( meta.isUseSegment() ) {
+        if ( meta.isUseCustomSegment() ) {
+          query.setSegment( environmentSubstitute( meta.getCustomSegment() ) );
+        } else {
+          query.setSegment( meta.getSegmentId() );
```

Comment:
The else clause is when you've clicked "Get Segments" and have chosen from a dropdown list